blueprints/app/files/app/index.html

```diff
@@ -15,6 +15,10 @@
     {{content-for "head-footer"}}
   </head>
   <body>
+    <noscript>
+      JavaScript must be enabled to run this application.
```

Comment:
> Does it not require putting checks before any DOM operations in hooks like didInsertElement?

Hooks like `didInsertElement`, `willInsertElement` which are meant for DOM manipulation don't run in FastBoot. Therefore you don't need checks in those hooks. See: http://ember-fastboot.com/docs/user-guide#designing-components

> But at the same time, if it was wrapped in something like:

That's exactly what I am saying: it is not _easily_ possible since FastBoot is not a static check. It is a run-time check. You could have your app running in FastBoot and you will still end up with this `noscript` check because there is no way to have `{{unless isFastBoot}}` check. The only way to avoid the `noscript` check when running in FastBoot is:

1. Have a middleware that runs after fastboot-express-middleware and strips out the `noscript` using some regex
2. Have `fastboot` itself remove it before appending the serialized template in the index.html.

Both of the above means we are magically removing the `noscript` tag which is going to cause confusion IMO.